### PR TITLE
chore: simplify url parsing code in debugger

### PR DESCRIPTION
**Pull request by Hanaasagi**
### What does this PR do?

Related to https://github.com/oven-sh/bun/issues/7225

Small fix for running `bun --inspect=0.0.0.0` and get `https://debug.bun.sh/#0.0.0.0:6499/undefined`

**Before**
```
>>​ bun --inspect=127.0.0.1:9898/prefix index.ts
Inspect in browser:
  https://debug.bun.sh/#127.0.0.1:9898/hnqrskhiqnj

>>​ bun --inspect=127.0.0.1 index.ts
Inspect in browser:
  https://debug.bun.sh/#127.0.0.1:6499/undefined

>>​ bun --inspect=127.0.0.1/prefix  index.ts
Inspect in browser:
  https://debug.bun.sh/#127.0.0.1:6499/prefix

```

**Now**

```
>>​  bun --inspect=127.0.0.1:9898/prefix index.ts
Inspect in browser:
  https://debug.bun.sh/#127.0.0.1:9898/prefix


>>​ bun --inspect=127.0.0.1 index.ts
Inspect in browser:
  https://debug.bun.sh/#127.0.0.1:6499/tlv5z50m13c

>>​ bun --inspect=127.0.0.1/prefix  index.ts
Inspect in browser:
  https://debug.bun.sh/#127.0.0.1:6499/prefix

```

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?
```TypeScript
import { expect, it } from "bun:test";

const defaultHostname = "localhost";
const defaultPort = 6499;

function parseUrl(url: string): URL {
  try {
    if (!url) {
      return new URL(randomId(), `ws://${defaultHostname}:${defaultPort}/`);
    } else if (url.startsWith("/")) {
      return new URL(url, `ws://${defaultHostname}:${defaultPort}/`);
    } else if (/^\d+$/.test(url)) {
      return new URL(randomId(), `ws://${defaultHostname}:${url}/`);
    } else {
      const base = new URL(`ws://${url}`);
      if (base.port === "") {
        base.port = defaultPort.toString();
      }
      if (base.pathname !== "/") {
        return base;
      }
      return new URL(randomId(), base);
    }
  } catch {
    throw new TypeError(`Invalid hostname or URL: '${url}'`);
  }
}

function randomId() {
  return "random";
}

it("simple test", () => {
  expect(parseUrl("")).toStrictEqual(new URL("ws://localhost:6499/random"));
  expect(parseUrl("9898")).toStrictEqual(new URL("ws://localhost:9898/random"));
  expect(parseUrl("/prefix")).toStrictEqual(new URL("ws://localhost:6499/prefix"));

  expect(parseUrl("localhost")).toStrictEqual(new URL("ws://localhost:6499/random"));
  expect(parseUrl("localhost:9898")).toStrictEqual(new URL("ws://localhost:9898/random"));
  expect(parseUrl("localhost:9898/prefix")).toStrictEqual(new URL("ws://localhost:9898/prefix"));
  expect(parseUrl("localhost:9898/")).toStrictEqual(new URL("ws://localhost:9898/random"));

  expect(parseUrl("0.0.0.0")).toStrictEqual(new URL("ws://0.0.0.0:6499/random"));
  expect(parseUrl("127.0.0.1:9898")).toStrictEqual(new URL("ws://127.0.0.1:9898/random"));
  expect(parseUrl("127.0.0.1:9898/prefix")).toStrictEqual(new URL("ws://127.0.0.1:9898/prefix"));
  expect(parseUrl("127.0.0.1:9898/")).toStrictEqual(new URL("ws://127.0.0.1:9898/random"));

  expect(parseUrl("[::1]")).toStrictEqual(new URL("ws://[::1]:6499/random"));
  expect(parseUrl("[::1]:9898")).toStrictEqual(new URL("ws://[::1]:9898/random"));
  expect(parseUrl("[::1]:9898/prefix")).toStrictEqual(new URL("ws://[::1]:9898/prefix"));
  expect(parseUrl("[::1]:9898/")).toStrictEqual(new URL("ws://[::1]:9898/random"));
});



```

### CODE REVIEW

**Comment by Electroid** [comment]
Could we add those test cases to a new test file? We should have test coverage for debugger but don't currently. Maybe make `test/cli/debug`

**Comment by Hanaasagi** [comment]
> Could we add those test cases to a new test file? We should have test coverage for debugger but don't currently. Maybe make test/cli/debug

Done.

File: src/js/internal/debugger.ts [hunk 1]
```diff
@@ -264,30 +264,30 @@ function bufferedWriter(writer: Writer): Writer {
...
+      }
+      return new URL(randomId(), base);
     }
   } catch {
     throw new TypeError(`Invalid hostname or URL: '${url}'`);
   }
 }
 
-function randomId() {
+export function randomId() {
```
Comment (Jarred-Sumner) [inline src/js/internal/debugger.ts]: Neither of these functions should be exported. `export` breaks the build here. These can only be `export default` at the very bottom of the file.

Comment (Electroid) [reply src/js/internal/debugger.ts]: Instead, these can be tested with an integration test, e.g. running `bun --inspect={}`, then connecting to the WebSocket and verifying the URL.

**Comment by Hanaasagi** [comment]
no response for a long time, closed.

**Pull request closed without merging**